container/libcontainer/helpers.go

```diff
@@ -163,17 +159,42 @@ func scanInterfaceStats(netStatsFile string) ([]info.InterfaceStats, error) {
 
 	scanner := bufio.NewScanner(file)
 
+	// Discard header lines
+	for i := 0; i < 2; i++ {
+		b := scanner.Scan()
+		if !b {
+			return stats, scanner.Err()
+		}
+	}
+
 	for scanner.Scan() {
 		line := scanner.Text()
 		line = strings.Replace(line, ":", "", -1)
 
-		i := info.InterfaceStats{}
+		fields := strings.Fields(line)
+		// If the format of the  line is invalid then don't trust any of the stats
+		// in this file.
+		if len(fields) != 17 {
+			return []info.InterfaceStats{}, fmt.Errorf("invalid interface stats line: %v", line)
+		}
+
+		devName := fields[0]
+		if !isIgnoredDevice(devName) {
```

Comment:
I'd invert this condition and use `continue` instead. This will mean one less indentation level for the rest of the code. See https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow